### PR TITLE
Add ErgoStrafer keyboard

### PR DESCRIPTION
**Pull request by infinityis**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/custommk/ergostrafer/ergostrafer.h
Comment (waffle87) [inline keyboards/custommk/ergostrafer/ergostrafer.h:1]: This file doesn't seem to be needed.

File: keyboards/custommk/ergostrafer/readme.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# ergostrafer
+
+![ergostrafer](https://i.imgur.com/ncJgD3j.jpeg)
```
Comment (waffle87) [inline keyboards/custommk/ergostrafer/readme.md:3]:
```suggestion
![ergostrafer](https://i.imgur.com/ncJgD3jh.jpeg)
```

File: keyboards/custommk/ergostrafer/readme.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+ErgoStrafer is a gaming mechanical keyboard that reproduces the layout of the discontinued SteelSeries Merc Stealth a.k.a. Zboard.
+
+* Keyboard Maintainer: [customMK](https://github.com/customMK)
+* Hardware Supported: ErgoStrafer
+* Hardware Availability: [customMK](https://shop.custommk.com/collections/ergostrafer/products/ergostrafer)
+
+Make example for this keyboard (after setting up your build environment):
+
+    make custommk/ergostrafer:default
```
Comment (waffle87) [inline keyboards/custommk/ergostrafer/readme.md:13]:
```suggestion
    make custommk/ergostrafer:default
    
Flashing example for this keyboard:

    make custommk/ergostrafer:default:flash
```

File: keyboards/custommk/ergostrafer/rules.mk [hunk 3]
```diff
@@ -0,0 +1,4 @@
+EEPROM_DRIVER = spi
+
+AUDIO_ENABLE = yes           # Audio output
```
Comment (waffle87) [inline keyboards/custommk/ergostrafer/rules.mk:3]:
```suggestion
```

File: keyboards/custommk/ergostrafer/info.json [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    "bootloader": "stm32-dfu",
+    "diode_direction": "ROW2COL",
+    "features": {
+        "bootmagic": true,
+        "command": false,
+        "console": false,
+        "extrakey": true,
+        "mousekey": true,
+        "nkro": true,
+        "encoder": true
```
Comment (waffle87) [inline keyboards/custommk/ergostrafer/info.json:14]:
```suggestion
        "encoder": true,
        "audio": true
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/custommk/ergostrafer/rules.mk (hunk 3 above)
Comment (infinityis) [reply keyboards/custommk/ergostrafer/rules.mk]: I had missed that audio enabling could now be in info.json (wasn't in documentation yet), fixed.

**Review by waffle87 — APPROVED** [review]

File: keyboards/custommk/ergostrafer/mcuconf.h [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include_next <mcuconf.h>
+
+// Used for audio (?)
+#undef STM32_PWM_USE_TIM4 //timer 4 channel ?
```
Comment (drashna) [inline keyboards/custommk/ergostrafer/mcuconf.h:22]: Yes. PWMD4 uses Timer 4. Channel isn't specified here, but via the config.
```suggestion
#undef STM32_PWM_USE_TIM4
```

**Review by drashna — APPROVED** [review]

File: keyboards/custommk/ergostrafer/mcuconf.h (hunk 5 above)
Comment (infinityis) [reply keyboards/custommk/ergostrafer/mcuconf.h]: Yep, was just a leftover comment from earlier, I thought I had deleted it (whereas now I know I've deleted it)

**Pull request merged**